modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -0,0 +1,616 @@
+        break;
+    }
+}
+
+cv::gimpl::GOAKExecutable::GOAKExecutable(const ade::Graph& g,
+                                          const cv::GCompileArgs &args,
+                                          const std::vector<ade::NodeHandle>& nodes,
+                                          const std::vector<cv::gimpl::Data>& ins_data,
+                                          const std::vector<cv::gimpl::Data>& outs_data)
+    : m_g(g), m_gm(m_g), m_args(args),
+      m_device(nullptr), m_pipeline(new dai::Pipeline)
+    {
+        // FIXME: change the hard-coded behavior (XLinkIn path)
+        auto camRgb = m_pipeline->create<dai::node::ColorCamera>();
+        // FIXME: extract camera compile arguments here and properly convert them for dai
+        camRgb->setBoardSocket(dai::CameraBoardSocket::RGB);
+        camRgb->setResolution(dai::ColorCameraProperties::SensorResolution::THE_1080_P);
+
+        // Set camera output. Fixme: consider working with other camera outputs
+        m_camera_input = camRgb;
+        // FIXME: change when other camera censors are introduced
+        m_camera_wh = camRgb->getVideoSize();
+
+        // Prepare XLinkOut nodes for each output object in graph
+        for (size_t i = 0; i < outs_data.size(); ++i) {
+            auto xout = m_pipeline->create<dai::node::XLinkOut>();
+            std::string xout_name = "xout" + std::to_string(i);
+            m_out_queue_names.push_back(xout_name);
+            xout->setStreamName(xout_name);
+            m_xlink_outputs.push_back(xout);
+        }
+
+        // Create OAK node for each node in this backend
+        for (const auto& nh : nodes) {
+            if (m_gm.metadata(nh).get<NodeType>().t == NodeType::OP) {
+                const auto& op = m_gm.metadata(nh).get<Op>();
+                const auto &u = ConstOAKGraph(m_g).metadata(nh).get<OAKComponent>();
+                // pass kernel input args and compile args to prepare OAK node and
+                // store it to link later
+                m_oak_nodes[nh] = nullptr;
+                m_oak_node_inputs[nh] = {};
+                m_oak_node_outputs[nh] = {};
+
+                std::vector<cv::GArg> in_ctx_args;
+                in_ctx_args.reserve(op.args.size());
+                for (auto &op_arg : op.args) in_ctx_args.push_back(packArg(op_arg,
+                                                                           m_oak_node_inputs[nh],
+                                                                           m_oak_wrapped_args));
+                for (auto &&op_out : op.outs) outArg(op_out, m_oak_node_outputs[nh]);
+                GAPI_Assert(!m_oak_node_inputs[nh].empty());
+                GAPI_Assert(!m_oak_node_outputs[nh].empty());
+
+                GOAKContext ctx(m_pipeline, m_camera_wh, in_ctx_args, m_oak_node_outputs[nh]);
+                u.k.m_f(ctx, m_oak_nodes[nh], m_in_queues);
+                GAPI_Assert(m_oak_nodes[nh] != nullptr);
```

Comment:
it seems your `m_f` produces this `m_oak_nodes[nh]` value - y not make it a return value for the callback? so you could write like
```c++
m_oak_nodes[nh] = u.k.m_f(ctx, m_in_queues);`
```
Also: is it 100% clear we have 1:1 correspondence between DAI nodes and our G-API node handles? Can a single Op in G-API unroll into multiple DAI ops?

Reply:
Didn't quite get the second part. Is it like ImageManip in DAI unrolls into crop, resize and cvcColor in G-API but the question is - do we have reversed cases? I think no, but I need to investigate further

Reply:
Done with the first part

Reply:
Think of "compound kernels" for the 2nd part.